### PR TITLE
Blog post: Elm and Electron

### PR DESCRIPTION
**Pull request by nirgn975**
#471

### CODE REVIEW

**Comment by nirgn975** [comment]
@amitaibu Ready for review (Don't know about the thumb, but it's the best thing I come up with).

File: app/_posts/2016-07-28-elm-with-electron.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+{% include setup %}
+
+I work at Gizra, so it was only a matter of time before the Elm thing infected me as well, and I think it's growing on me.
+
+I wanted to start playing with it, to build something new, not just the plain old TodoMVC. So, the idea was to build a YouTube playlist app, so everyone can share their YouTube music links. First thing first, I took the TodoMVC in Elm and got it to work in Electron.
+
+[Electron](http://electron.atom.io/) enables you to build cross platform desktop apps with web technologies.
+
+In my mind we should get couple of things out of the box:
```
Comment (RachelBaram) [inline app/_posts/2016-07-28-elm-with-electron.md]: @nirgn975 What are you trying to say here? I was not sure how to edit this line.

File: app/_posts/2016-07-28-elm-with-electron.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+ * Compile SASS to css.
+ * Compile elm to JS.
+ * Watch and auto-reload.
+ * Automagically download and install elm packages.
```
Comment (RachelBaram) [inline app/_posts/2016-07-28-elm-with-electron.md]: Is "Automagically" a technical word or did you mean "Automatically"?

File: app/_posts/2016-07-28-elm-with-electron.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * Start the electron app.
+
+Just to make things clear, the only things I was willing to do were `git clone ..`, `npm install`, and `gulp`.
```
Comment (RachelBaram) [inline app/_posts/2016-07-28-elm-with-electron.md]: What do you mean by "the only things I was willing to do"? This sentence needs to be edited to better say what you intended.

File: app/_posts/2016-07-28-elm-with-electron.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+<!-- more -->
+
+My project is called, wait for it, `YouTube-Playlist` (mind blowing). In Gizra we achieve what we aim for and then open source it, because why not. So, of course, you're more than welcome to fork the [GitHub repo](https://github.com/nirgn975/YouTube-Playlist).
+
+After you `git clone git@github.com:nirgn975/YouTube-Playlist.git`, and `npm install`, just run `gulp` and see the magic happen:
+
+{% include thumbnail.html image_path="assets/images/posts/elm-electron-build/gulp-electron.jpg" caption="Gulp install Elm packages automagically and start Electron" %}
```
Comment (RachelBaram) [inline app/_posts/2016-07-28-elm-with-electron.md]: Same question about "automagically"

File: app/_posts/2016-07-28-elm-with-electron.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+---
+title: "Elm - if you can't beat them, join them"
+tags:
+  - Elm
+  - Electron
```
Comment (RachelBaram) [inline app/_posts/2016-07-28-elm-with-electron.md]: @nirgn975  I am working to make our tags list meaningful.  Do you see "Electron" as something someone would actually check out what else was written about it? If not, I would prefer that we take it out.  Also, are any of our other [existing tags](http://www.gizra.com/tags.html) fit this post? We are working to label each of our content pieces with 2-4 (meaningful) tags. Thanks!

File: app/_posts/2016-07-28-elm-with-electron.md (hunk 2 above)
Comment (nirgn975) [inline app/_posts/2016-07-28-elm-with-electron.md]: @RachelBaram I meant "Automagically" (:

Comment (RachelBaram) [inline app/_posts/2016-07-28-elm-with-electron.md]: I learn something new every day.

File: app/_posts/2016-07-28-elm-with-electron.md (hunk 1 above)
Comment (nirgn975) [inline app/_posts/2016-07-28-elm-with-electron.md]: I try to say that when we start the project, we should get things without any work ("out of the box"), those things are listed below.

File: app/_posts/2016-07-28-elm-with-electron.md (hunk 5 above)
Comment (nirgn975) [inline app/_posts/2016-07-28-elm-with-electron.md]: @RachelBaram Yes, Electron is a tool/technology and if someone interested in it he would probably write this word in his google query. From this list, maybe `open source` would fit.

File: app/_posts/2016-07-28-elm-with-electron.md (hunk 2 above)
Comment (nirgn975) [inline app/_posts/2016-07-28-elm-with-electron.md]: [automagically definition](https://www.google.co.il/search?espv=2&q=automagically+definition&oq=Automagically+d&gs_l=serp.3.0.0j0i22i30l6.3982.4474.0.5210.2.2.0.0.0.0.136.266.0j2.2.0....0...1c.1.64.serp..0.2.265.Dwtw6Nb0e3Y). All of learn every day.

File: app/_posts/2016-07-28-elm-with-electron.md (hunk 3 above)
Comment (nirgn975) [inline app/_posts/2016-07-28-elm-with-electron.md]: There are a lot of installation and configuration in this project that needs to be done before you even start to write code, but I was not willing to do them every time. I write an automation script once so I (and everyone who use this project) will have to write only 3 commands before we can start the real work.

Those are the bare minimum commands I was willing to accept. If I think I can remove another command, I wouldn't write this post, yet.

**Comment by nirgn975** [comment]
@amitaibu Ready for review.

File: app/_posts/2016-07-28-elm-with-electron.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+{% include thumbnail.html image_path="assets/images/posts/elm-electron-build/gulp-electron.jpg" caption="Gulp install Elm packages automagically and start Electron" %}
+
+After gulp does its magic, a new window will open:
+
+{% include thumbnail.html image_path="assets/images/posts/elm-electron-build/TodoMVC.jpg" caption="Elm code run inside Electron" %}
+
+Tada! Your Elm app runs inside Electron. And as I promised, all of the above just happens for you.
+
+Also, to help you guys start develop, I set it up so [the devtools will open](https://github.com/nirgn975/Elmctron/blob/master/main-electron.js#L31-L32) when the app fires up.
```
Comment (amitaibu) [inline app/_posts/2016-07-28-elm-with-electron.md]: `you guys` => `folks` (as you know, there could be girls as-well 😉 )

**Comment by amitaibu** [comment]
I've optimized images (and reduced their size. It shouldn't be more than 800px width)

Merged and deployed - http://www.gizra.com/content/elm-electron-build/

Thanks!

**Pull request merged**